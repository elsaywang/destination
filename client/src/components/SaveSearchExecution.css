:local(.trigger) {
    margin: 2.75em 0 0 0.25em;
}
:local(.triggerDialog) {
    padding: 1em 1em 1em 2em;
}
:local(.contentCheckBox) {
    margin: 0.5em 0 0 0.75em;
}
:local(.contentRadioGroup) {
    margin-top: 0.5em;
}
