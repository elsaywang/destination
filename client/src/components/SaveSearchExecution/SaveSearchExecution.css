:local(.triggerDialog) {
    padding: 1em 1em 1em 2em;
}
:local(.contentTextField) {
    min-width: 230px;
}
:local(.contentCheckBox) {
    margin: 0.5em 0 0 0.75em;
}
:local(.contentRadioGroup) {
    margin-top: 0.5em;
}
