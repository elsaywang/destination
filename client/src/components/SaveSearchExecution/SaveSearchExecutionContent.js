import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Label from '../../components/common/Label';
import Textfield from '@react/react-spectrum/Textfield';
import Checkbox from '@react/react-spectrum/Checkbox';
import RadioGroup from '@react/react-spectrum/RadioGroup';
import Radio from '@react/react-spectrum/Radio';
import Select from '@react/react-spectrum/Select';
import styles from './SaveSearchExecution.css';
import {
    textFieldLabelName,
    textFieldPlaceHolder,
    saveSearch,
    checkBoxLabel,
    defaultSorting,
} from './saveSearchExecutionMessages.js';
import {
    descending,
    sortingOptions,
    radioGroupOptions,
    defaultSortingKey,
} from '../../constants/sortOptions';

class SaveSearchExecutionContent extends Component {
    state = {
        isTrackInDashboardChecked: false,
    };

    toggleCheckbox = isTrackInDashboardChecked => {
        this.setState({ isTrackInDashboardChecked });
        this.props.onTrackResultInDashboardChange(isTrackInDashboardChecked);
    };

    handleRadioChange = value => {
        this.props.onSortingOrderChange(value === descending);
    };

    render() {
        return (
            <div data-test="save-this-search-dialog-content">
                <Label value={textFieldLabelName} labelFor={saveSearch}>
                    <Textfield
                        data-test="save-this-search-name-field"
                        className={styles.contentTextField}
                        placeholder={textFieldPlaceHolder}
                        onChange={this.props.onSaveSearchNameChange}
                        quiet
                    />
                </Label>
                <div className={styles.contentCheckBox}>
                    <Checkbox
                        data-test="save-this-search-checkbox"
                        label={checkBoxLabel}
                        onChange={this.toggleCheckbox}
                        checked={this.state.isTrackInDashboardChecked}
                    />
                </div>
                <Label value={defaultSorting} labelFor={defaultSorting}>
                    <Select
                        options={sortingOptions}
                        onChange={this.props.onDefaultSortingChange}
                        defaultValue={defaultSortingKey}
                    />
                    <div className={styles.contentRadioGroup}>
                        <RadioGroup>
                            {radioGroupOptions.map(({ label, value }) => (
                                <Radio
                                    key={value}
                                    {...{ label, value }}
                                    checked={value === descending}
                                    onChange={this.handleRadioChange}
                                />
                            ))}
                        </RadioGroup>
                    </div>
                </Label>
            </div>
        );
    }
}

SaveSearchExecutionContent.propTypes = {
    onSaveSearchNameChange: PropTypes.func,
    onTrackResultInDashboardChange: PropTypes.func,
    onDefaultSortingChange: PropTypes.func,
    onSortingOrderChange: PropTypes.func,
};
export default SaveSearchExecutionContent;
