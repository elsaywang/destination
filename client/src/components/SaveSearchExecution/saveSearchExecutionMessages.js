export const saveThisSearch = 'Save This Search';
export const checkBoxLabel = 'Track this search result in dashboard';
export const defaultSorting = 'Default Sorting';
export const textFieldLabelName = 'Name';
export const textFieldPlaceHolder = 'Enter a name for this saved search';
