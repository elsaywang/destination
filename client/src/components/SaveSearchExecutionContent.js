import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import Label from '../components/common/Label';
import Textfield from '@react/react-spectrum/Textfield';
import Checkbox from '@react/react-spectrum/Checkbox';
import SaveSearchSortingCriteria from './SaveSearchSortingCriteria';
import styles from './SaveSearchExecution.css';
import {
    textFieldLabelName,
    textFieldPlaceHolder,
    saveThisSearch,
    checkBoxLabel,
    defaultSorting,
    radioGroupOptions,
    sortingOptions,
} from '../utils/saveSearch';

class SaveSearchExecutionContent extends Component {
    constructor() {
        super();
        this.state = {
            isBoxChecked: false,
        };
    }

    toggleCheckbox = () => {
        this.setState({ isBoxChecked: !this.state.isBoxChecked });
        this.props.onTrackResultInDashBoardChange(!this.state.isBoxChecked);
    };

    handleRadioChange = value => {
        this.props.onSortingOrderChange(value === 'descending');
    };

    executionForm = () => {
        return (
            <Fragment>
                <Label value={textFieldLabelName} labelFor={saveThisSearch}>
                    <Textfield
                        style={{ minWidth: 230 }}
                        placeholder={textFieldPlaceHolder}
                        onChange={this.props.onSaveSearchNameChange}
                        quiet
                    />
                </Label>
                <div className={styles.contentCheckBox}>
                    <Checkbox
                        label={checkBoxLabel}
                        onChange={this.toggleCheckbox}
                        checked={this.state.isBoxChecked}
                    />
                </div>
                {this.state.isBoxChecked && (
                    <Label value={defaultSorting} labelFor={defaultSorting}>
                        <SaveSearchSortingCriteria
                            selectOptions={sortingOptions}
                            radioGroupOptions={radioGroupOptions}
                            onDefaultSortingChange={this.props.onDefaultSortingChange}
                            handleRadioChange={this.handleRadioChange}
                            radioCheckedValue="descending"
                            radioGroupStyle={styles.contentRadioGroup}
                        />
                    </Label>
                )}
            </Fragment>
        );
    };
    render() {
        return this.executionForm();
    }
}

SaveSearchExecutionContent.propTypes = {
    onSaveSearchNameChange: PropTypes.func,
    onTrackResultInDashBoardChange: PropTypes.func,
    onDefaultSortingChange: PropTypes.func,
    onSortingOrderChange: PropTypes.func,
};
export default SaveSearchExecutionContent;
