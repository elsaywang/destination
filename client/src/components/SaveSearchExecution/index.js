import SaveSearchExecution from './SaveSearchExecution';

export default SaveSearchExecution;
