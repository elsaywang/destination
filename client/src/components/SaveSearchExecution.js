import React, { Component } from 'react';
import Button from '@react/react-spectrum/Button';
import PropTypes from 'prop-types';
import Add from '@react/react-spectrum/Icon/Add';
import styles from './SaveSearchExecution.css';
// TODO: BRING BACK OverlayTrigger once PopOver is fixed
// import OverlayTrigger from '@react/react-spectrum/OverlayTrigger';
// import Popover from '@react/react-spectrum/Popover';
import ModalTrigger from '@react/react-spectrum/ModalTrigger';
import Dialog from '@react/react-spectrum/Dialog';
import SaveSearchExecutionContent from './SaveSearchExecutionContent';
import { saveThisSearch } from '../utils/saveSearch';

class SaveSearchExecution extends Component {
    render() {
        const {
            confirmSaveThisSearch,
            updateSaveSearchName,
            cancleSaveSearch,
            trackSearchResultInDashboard,
            selectDefaultSorting,
            changeSortingOrder,
        } = this.props;
        return (
            // TODO: switch back to OverlayTrigger once PopOver is fixed
            // <div className={styles.trigger}>
            //     <OverlayTrigger trigger="click" placement="bottom" onHide={cancleSaveSearch}>
            //         <Button label={saveThisSearch} variant="action" quiet icon={<Add />} />
            //         <Popover className={styles.triggerDialog}>
            //             <SaveSearchExecutionContent
            //                 onSaveSearchNameChange={updateSaveSearchName}
            //                 onTrackResultInDashBoardChange={trackSearchResultInDashboard}
            //                 onDefaultSortingChange={selectDefaultSorting}
            //                 onSortingOrderChange={changeSortingOrder}
            //             />
            //         </Popover>
            //     </OverlayTrigger>
            // </div>
            <div className={styles.trigger}>
                <ModalTrigger>
                    <Button label={saveThisSearch} variant="action" quiet icon={<Add />} />
                    <Dialog
                        className={styles.triggerDialog}
                        modalcontent
                        confirmLabel="Save"
                        size="S"
                        variant="information"
                        cancelLabel="Cancel"
                        onConfirm={confirmSaveThisSearch}
                        onCancel={cancleSaveSearch}>
                        <SaveSearchExecutionContent
                            onSaveSearchNameChange={updateSaveSearchName}
                            onTrackResultInDashBoardChange={trackSearchResultInDashboard}
                            onDefaultSortingChange={selectDefaultSorting}
                            onSortingOrderChange={changeSortingOrder}
                        />
                    </Dialog>
                </ModalTrigger>
            </div>
        );
    }
}

SaveSearchExecution.propTypes = {
    confirmSaveThisSearch: PropTypes.func,
    cancleSaveSearch: PropTypes.func,
    updateSaveSearchName: PropTypes.func,
    trackSearchResultInDashboard: PropTypes.func,
    selectDefaultSorting: PropTypes.func,
    changeSortingOrder: PropTypes.func,
};

export default SaveSearchExecution;
