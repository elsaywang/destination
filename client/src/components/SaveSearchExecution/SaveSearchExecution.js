import React, { Component } from 'react';
import Button from '@react/react-spectrum/Button';
import PropTypes from 'prop-types';
import Add from '@react/react-spectrum/Icon/Add';
import styles from './SaveSearchExecution.css';
// TODO: BRING BACK OverlayTrigger once PopOver is fixed
// import OverlayTrigger from '@react/react-spectrum/OverlayTrigger';
// import Popover from '@react/react-spectrum/Popover';
import ModalTrigger from '@react/react-spectrum/ModalTrigger';
import Dialog from '@react/react-spectrum/Dialog';
import SaveSearchExecutionContent from './SaveSearchExecutionContent';
import { saveSearch } from './saveSearchExecutionMessages';

class SaveSearchExecution extends Component {
    render() {
        const {
            disabled,
            confirmSaveThisSearch,
            updateSaveSearchName,
            cancelSaveSearch,
            trackSearchResultInDashboard,
            selectDefaultSorting,
            changeSortingOrder,
        } = this.props;
        return (
            // TODO: switch back to OverlayTrigger once PopOver is fixed
            //     <OverlayTrigger trigger="click" placement="bottom" onHide={cancelSaveSearch}>
            //         <Button label={saveSearch} variant="action" quiet icon={<Add />} />
            //         <Popover className={styles.triggerDialog}>
            //             <SaveSearchExecutionContent
            //                 onSaveSearchNameChange={updateSaveSearchName}
            //                 onTrackResultInDashboardChange={trackSearchResultInDashboard}
            //                 onDefaultSortingChange={selectDefaultSorting}
            //                 onSortingOrderChange={changeSortingOrder}
            //             />
            //         </Popover>
            //     </OverlayTrigger>
            <ModalTrigger>
                <Button
                    data-test="save-this-search-button"
                    label={saveSearch}
                    variant="action"
                    quiet
                    icon={<Add />}
                    disabled={disabled}
                />
                <Dialog
                    className={styles.triggerDialog}
                    modalcontent
                    confirmLabel="Save"
                    size="S"
                    variant="information"
                    cancelLabel="Cancel"
                    onConfirm={confirmSaveThisSearch}
                    onCancel={cancelSaveSearch}>
                    <SaveSearchExecutionContent
                        onSaveSearchNameChange={updateSaveSearchName}
                        onTrackResultInDashboardChange={trackSearchResultInDashboard}
                        onDefaultSortingChange={selectDefaultSorting}
                        onSortingOrderChange={changeSortingOrder}
                    />
                </Dialog>
            </ModalTrigger>
        );
    }
}

SaveSearchExecution.propTypes = {
    disabled: PropTypes.bool,
    confirmSaveThisSearch: PropTypes.func,
    cancelSaveSearch: PropTypes.func,
    updateSaveSearchName: PropTypes.func,
    trackSearchResultInDashboard: PropTypes.func,
    selectDefaultSorting: PropTypes.func,
    changeSortingOrder: PropTypes.func,
};

export default SaveSearchExecution;
