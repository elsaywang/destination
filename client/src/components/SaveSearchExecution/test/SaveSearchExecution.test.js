import React, { Component } from 'react';
import Button from '@react/react-spectrum/Button';
import { shallow } from 'enzyme';
import SaveSearchExecution from '../SaveSearchExecution';
import SaveSearchExecutionContent from '../SaveSearchExecutionContent';
import OverlayTrigger from '@react/react-spectrum/OverlayTrigger';
import ModalTrigger from '@react/react-spectrum/ModalTrigger';
import Dialog from '@react/react-spectrum/Dialog';
import Popover from '@react/react-spectrum/Popover';
import Add from '@react/react-spectrum/Icon/Add';

describe('<SaveSearchExecution/>', () => {
    const wrapper = shallow(<SaveSearchExecution />);
    const children = wrapper.find(ModalTrigger).children();
    it('matches the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('renders <ModalTrigger/>', () => {
        expect(wrapper.find(ModalTrigger).exists()).toBeTruthy();
    });

    it('<ModalTrigger/> including children <Button/> and <Dialog/>', () => {
        expect(children.find(Button).exists()).toBeTruthy();
        expect(children.find(Dialog).exists()).toBeTruthy();
    });

    it('child <Dialog/> includes corrent content <SaveSearchExecutionContent/>', () => {
        expect(
            children
                .find(Dialog)
                .find(SaveSearchExecutionContent)
                .exists(),
        ).toBeTruthy();
    });
});

//TODO:ADD BACK once OverlayTrigger and PopOver are updated
xdescribe('<SaveSearchExecution/>', () => {
    const wrapper = shallow(<SaveSearchExecution />);
    const children = wrapper.find(OverlayTrigger).children();
    it('matches the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('renders <OverlayTrigger/>', () => {
        expect(wrapper.find(OverlayTrigger).exists()).toBeTruthy();
    });

    it('<OverlayTrigger/> including children <Button/> and <Popover/>', () => {
        expect(children.find(Button).exists()).toBeTruthy();
        expect(children.find(Popover).exists()).toBeTruthy();
    });

    it('child <Popover/> includes corrent content <SaveSearchExecutionContent/>', () => {
        expect(
            children
                .find(Popover)
                .find(SaveSearchExecutionContent)
                .exists(),
        ).toBeTruthy();
    });
});
