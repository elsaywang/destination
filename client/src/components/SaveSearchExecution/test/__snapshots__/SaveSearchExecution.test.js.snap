// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SaveSearchExecution/> rendering when error passes in matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SaveSearchExecution
    error={
        Object {
            "errorMessage": "Invalid Method",
            "hasError": true,
          }
    }
    isSavedSearchLimitReached={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "errorMessage": "Invalid Method",
      "isInvalid": true,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "errorMessage": "Invalid Method",
        "isInvalid": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<SaveSearchExecution/> rendering when no error passes in matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SaveSearchExecution
    cancelSaveSearch={[Function]}
    changeSortingOrder={[Function]}
    confirmSaveThisSearch={[Function]}
    error={
        Object {
            "errorMessage": "",
            "hasError": false,
          }
    }
    isSavedSearchLimitReached={true}
    savedSearchLimit={10}
    selectDefaultSorting={[Function]}
    trackSearchResultInDashboard={[Function]}
    updateSaveSearchName={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div
          data-test="save-this-search-button-overlay-trigger"
>
          <OverlayTrigger
                    boundariesElement={[Function]}
                    crossOffset={0}
                    defaultShow={false}
                    delayHide={100}
                    disabled={false}
                    flip={true}
                    offset={0}
                    placement="top"
                    trigger={
                              Array [
                                        "hover",
                                        "focus",
                                      ]
                    }
          >
                    <Button
                              autoFocus={false}
                              block={false}
                              data-test="save-this-search-button"
                              disabled={true}
                              element="button"
                              icon={<Add />}
                              invalid={false}
                              label="Save This Search"
                              logic={false}
                              quiet={true}
                              selected={false}
                              variant="action"
                    />
                    <Tooltip
                              data-test="saved-search-limit-message"
                              open={true}
                              placement="right"
                              role="tooltip"
                              variant="inspect"
                    >
                              Saved Search limit (10) reached.
                    </Tooltip>
          </OverlayTrigger>
</div>,
        <Dialog
          autoFocusButton={null}
          backdropClickable={false}
          cancelLabel="Cancel"
          className="triggerDialog"
          confirmDisabled={false}
          confirmLabel="Save"
          keyboardConfirm={false}
          modalcontent={true}
          mode="centered"
          onCancel={[Function]}
          onClose={[Function]}
          onConfirm={[Function]}
          open={true}
          role="dialog"
          size="S"
          variant="information"
>
          <SaveSearchExecutionContent
                    onDefaultSortingChange={[Function]}
                    onSaveSearchNameChange={[Function]}
                    onSortingOrderChange={[Function]}
                    onTrackResultInDashboardChange={[Function]}
          />
</Dialog>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <OverlayTrigger
            boundariesElement={[Function]}
            crossOffset={0}
            defaultShow={false}
            delayHide={100}
            disabled={false}
            flip={true}
            offset={0}
            placement="top"
            trigger={
                        Array [
                                    "hover",
                                    "focus",
                                  ]
            }
>
            <Button
                        autoFocus={false}
                        block={false}
                        data-test="save-this-search-button"
                        disabled={true}
                        element="button"
                        icon={<Add />}
                        invalid={false}
                        label="Save This Search"
                        logic={false}
                        quiet={true}
                        selected={false}
                        variant="action"
            />
            <Tooltip
                        data-test="saved-search-limit-message"
                        open={true}
                        placement="right"
                        role="tooltip"
                        variant="inspect"
            >
                        Saved Search limit (10) reached.
            </Tooltip>
</OverlayTrigger>,
          "data-test": "save-this-search-button-overlay-trigger",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "boundariesElement": [Function],
            "children": Array [
              <Button
                autoFocus={false}
                block={false}
                data-test="save-this-search-button"
                disabled={true}
                element="button"
                icon={<Add />}
                invalid={false}
                label="Save This Search"
                logic={false}
                quiet={true}
                selected={false}
                variant="action"
/>,
              <Tooltip
                data-test="saved-search-limit-message"
                open={true}
                placement="right"
                role="tooltip"
                variant="inspect"
>
                Saved Search limit (10) reached.
</Tooltip>,
            ],
            "crossOffset": 0,
            "defaultShow": false,
            "delayHide": 100,
            "disabled": false,
            "flip": true,
            "offset": 0,
            "placement": "top",
            "trigger": Array [
              "hover",
              "focus",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "autoFocus": false,
                "block": false,
                "data-test": "save-this-search-button",
                "disabled": true,
                "element": "button",
                "icon": <Add />,
                "invalid": false,
                "label": "Save This Search",
                "logic": false,
                "quiet": true,
                "selected": false,
                "variant": "action",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Saved Search limit (10) reached.",
                "data-test": "saved-search-limit-message",
                "open": true,
                "placement": "right",
                "role": "tooltip",
                "variant": "inspect",
              },
              "ref": null,
              "rendered": "Saved Search limit (10) reached.",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocusButton": null,
          "backdropClickable": false,
          "cancelLabel": "Cancel",
          "children": <SaveSearchExecutionContent
            onDefaultSortingChange={[Function]}
            onSaveSearchNameChange={[Function]}
            onSortingOrderChange={[Function]}
            onTrackResultInDashboardChange={[Function]}
/>,
          "className": "triggerDialog",
          "confirmDisabled": false,
          "confirmLabel": "Save",
          "keyboardConfirm": false,
          "modalcontent": true,
          "mode": "centered",
          "onCancel": [Function],
          "onClose": [Function],
          "onConfirm": [Function],
          "open": true,
          "role": "dialog",
          "size": "S",
          "variant": "information",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onDefaultSortingChange": [Function],
            "onSaveSearchNameChange": [Function],
            "onSortingOrderChange": [Function],
            "onTrackResultInDashboardChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            data-test="save-this-search-button-overlay-trigger"
>
            <OverlayTrigger
                        boundariesElement={[Function]}
                        crossOffset={0}
                        defaultShow={false}
                        delayHide={100}
                        disabled={false}
                        flip={true}
                        offset={0}
                        placement="top"
                        trigger={
                                    Array [
                                                "hover",
                                                "focus",
                                              ]
                        }
            >
                        <Button
                                    autoFocus={false}
                                    block={false}
                                    data-test="save-this-search-button"
                                    disabled={true}
                                    element="button"
                                    icon={<Add />}
                                    invalid={false}
                                    label="Save This Search"
                                    logic={false}
                                    quiet={true}
                                    selected={false}
                                    variant="action"
                        />
                        <Tooltip
                                    data-test="saved-search-limit-message"
                                    open={true}
                                    placement="right"
                                    role="tooltip"
                                    variant="inspect"
                        >
                                    Saved Search limit (10) reached.
                        </Tooltip>
            </OverlayTrigger>
</div>,
          <Dialog
            autoFocusButton={null}
            backdropClickable={false}
            cancelLabel="Cancel"
            className="triggerDialog"
            confirmDisabled={false}
            confirmLabel="Save"
            keyboardConfirm={false}
            modalcontent={true}
            mode="centered"
            onCancel={[Function]}
            onClose={[Function]}
            onConfirm={[Function]}
            open={true}
            role="dialog"
            size="S"
            variant="information"
>
            <SaveSearchExecutionContent
                        onDefaultSortingChange={[Function]}
                        onSaveSearchNameChange={[Function]}
                        onSortingOrderChange={[Function]}
                        onTrackResultInDashboardChange={[Function]}
            />
</Dialog>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <OverlayTrigger
              boundariesElement={[Function]}
              crossOffset={0}
              defaultShow={false}
              delayHide={100}
              disabled={false}
              flip={true}
              offset={0}
              placement="top"
              trigger={
                            Array [
                                          "hover",
                                          "focus",
                                        ]
              }
>
              <Button
                            autoFocus={false}
                            block={false}
                            data-test="save-this-search-button"
                            disabled={true}
                            element="button"
                            icon={<Add />}
                            invalid={false}
                            label="Save This Search"
                            logic={false}
                            quiet={true}
                            selected={false}
                            variant="action"
              />
              <Tooltip
                            data-test="saved-search-limit-message"
                            open={true}
                            placement="right"
                            role="tooltip"
                            variant="inspect"
              >
                            Saved Search limit (10) reached.
              </Tooltip>
</OverlayTrigger>,
            "data-test": "save-this-search-button-overlay-trigger",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "boundariesElement": [Function],
              "children": Array [
                <Button
                  autoFocus={false}
                  block={false}
                  data-test="save-this-search-button"
                  disabled={true}
                  element="button"
                  icon={<Add />}
                  invalid={false}
                  label="Save This Search"
                  logic={false}
                  quiet={true}
                  selected={false}
                  variant="action"
/>,
                <Tooltip
                  data-test="saved-search-limit-message"
                  open={true}
                  placement="right"
                  role="tooltip"
                  variant="inspect"
>
                  Saved Search limit (10) reached.
</Tooltip>,
              ],
              "crossOffset": 0,
              "defaultShow": false,
              "delayHide": 100,
              "disabled": false,
              "flip": true,
              "offset": 0,
              "placement": "top",
              "trigger": Array [
                "hover",
                "focus",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "autoFocus": false,
                  "block": false,
                  "data-test": "save-this-search-button",
                  "disabled": true,
                  "element": "button",
                  "icon": <Add />,
                  "invalid": false,
                  "label": "Save This Search",
                  "logic": false,
                  "quiet": true,
                  "selected": false,
                  "variant": "action",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Saved Search limit (10) reached.",
                  "data-test": "saved-search-limit-message",
                  "open": true,
                  "placement": "right",
                  "role": "tooltip",
                  "variant": "inspect",
                },
                "ref": null,
                "rendered": "Saved Search limit (10) reached.",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoFocusButton": null,
            "backdropClickable": false,
            "cancelLabel": "Cancel",
            "children": <SaveSearchExecutionContent
              onDefaultSortingChange={[Function]}
              onSaveSearchNameChange={[Function]}
              onSortingOrderChange={[Function]}
              onTrackResultInDashboardChange={[Function]}
/>,
            "className": "triggerDialog",
            "confirmDisabled": false,
            "confirmLabel": "Save",
            "keyboardConfirm": false,
            "modalcontent": true,
            "mode": "centered",
            "onCancel": [Function],
            "onClose": [Function],
            "onConfirm": [Function],
            "open": true,
            "role": "dialog",
            "size": "S",
            "variant": "information",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onDefaultSortingChange": [Function],
              "onSaveSearchNameChange": [Function],
              "onSortingOrderChange": [Function],
              "onTrackResultInDashboardChange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
